src/NosCore.GameObject/ExchangeData.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using NosCore.GameObject.Services.ItemBuilder.Item;
+
+namespace NosCore.GameObject
+{
+    public class ExchangeData : ConcurrentDictionary<long, ItemInstance>
+    {
+        public long TargetVisualId { get; set; }
```

Comment:
useless the caller of exchange data should keep a dictionnary between targetId and data

Reply:
Yeah, but in the exchange request packet, there is sometimes no target id, and no data to fill the dictionnary

Reply:
so just make it nullable